### ISSUE TITLE
High package activation time

### ISSUE THREAD

**Issue by borishere** [abe33/atom-pigments#306]
Hi, sometimes I get high activation time as on screenshot. Is it expected or?

![image](https://cloud.githubusercontent.com/assets/4622497/26124888/d83ac33c-3a88-11e7-86ba-8dbcee313824.png)

**Comment by borishere** [comment]
In fact, when I launch Atom, I can work in it only after ~50 seconds (Atom is hanging).
Without 'pigments' package Atom is ready for work after ~7 seconds.

**Comment by C-Bouthoorn** [comment]
Could you please share your Atom version with us? If it is 1.18-beta, https://github.com/abe33/atom-pigments/issues/308 could be an issue here.

**Comment by borishere** [comment]
@C-Bouthoorn I use `1.18.0-beta2`.
I discovered that scanning for variables (color palette) causes this hanging and my project has a lot of color variables. Seems that nothing can help me until I disable searching for variables :(

**Comment by abe33** [comment]
Actually it probably might not be the scanning itself that takes time, as it's an async operation, but rather the deserialization of previously discovered variables. Because scanning a whole project directory takes time and resources the package is using the state serialization mechanism Atom provides to store these variables between sessions, but now it's the serialization/deserialization that become the bottleneck as it's a synchronous operation (I guess they're just `JSON.stringify` & `JSON.parse` the whole thing). 
I don't really have a plan to rewrite that part to use a different strategy as it'll imply a lot of new issue to tackle. Without knowledge of how is structured it's hard for me to give you a workaround, but generally, if if the project is really huge, with a lot of dependencies, there might be a lot of variables that doesn't come directly from your code but are found in dependencies, and in that case it sometime make sense to excludes these dependencies from pigments' paths to reduce the burden on state serialization.

**Comment by C-Bouthoorn** [comment]
Closing in favour of the [master post](https://github.com/abe33/atom-pigments/issues/351).
